### PR TITLE
feat:added check for science lab device and dynamic selection of wavetype

### LINKED ISSUE #1097: the waves are not properly set for both pins and the app is crashing
**Actual Behaviour**

The waves are not changing dynamically for both pins and the app is crashing when the device is not connected

**Expected Behaviour**

the wavetype for both pins W1 and W2 should change dynamically and check the sciencelab connection to prevent app crashing.

**Steps to reproduce it**

Wave Generator -> handle wavetype for both pins.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1097 

**Changes**:
- Added the check for science lab
- added toast if device is not connected
- the wavetype are selected for both waves separately.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [ ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**:[app-debug.zip](https://github.com/fossasia/pslab-android/files/2118773/app-debug.zip)

### CODE REVIEW

**Comment by Avjeet** [comment]
@harsh-2711 @Vikum94 @abhinavraj23 could u guys plz test this PR with your devices as my channel 2 is not working.

**Comment by akarshan96** [comment]
@Avjeet have you tried using CH2 individually?

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 1]
```diff
@@ -354,24 +356,27 @@ public void onClick(View view) {
...
+                            scienceLab.setW2(freq2, waveType2);
+                        } else {
+                            scienceLab.setWaves(freq1, phase, freq2);
+                        }
                     }
+                    Intent intent = new Intent(WaveGeneratorActivity.this, OscilloscopeActivity.class);
+                    intent.putExtra("who", "WaveGenerator");
+                    startActivity(intent);
+                } else {
+                    Toast.makeText(WaveGeneratorActivity.this, "PSLab device not connected", Toast.LENGTH_SHORT).show();
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: Let's use string resources here.

**Comment by Avjeet** [comment]
@akarshan I tried the connecting the CH2 with every pin, the CH1 is giving correct output but same wave is not showing on the CH2.

**Comment by akarshan96** [comment]
@Avjeet see if any exception is thrown while CH2 is used to read the output. But don't prioritize this, do it when you get time.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -858,5 +858,6 @@
     GND. GND is meant for Ground and you can use any of the PSLab device GND pins as they are common.\n\n
     \u2022 Then go to the Configure tab from the bottom navigation bar and choose the BH-1750 in the drop down
     menu Select Sensor.\n</string>
+    <string name="device_not_connected_warning">PSLab device not connected</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Similar string is available in 138-142 in `strings.xml`. Please reuse that one without creating new strings

Comment (Avjeet) [reply app/src/main/res/values/strings.xml]: ok

**Comment by Avjeet** [comment]
@CloudyPadmal done changes

**Comment by Avjeet** [comment]
@akarshan96 plz review this

**Pull request merged**